### PR TITLE
Feature. Controllers for database

### LINKED ISSUE #37: Backend: support controllers for database related work
We (with @petertrr) decided that we should have a single point to work with a database for all microservices.
For example: `orchestrator`, `backend-preprocessor` and `backend` should work with the same microservice that will do the same base-related work. 

So we need 2 endpoints for it:
1. Orchestrator (to receive post requests with Statuses)
2. Backend-preprocessor (to receive post requests with test IDs)

### PR DESCRIPTION
**Pull request by aktsay6**
### What's done:
  * Added schemes for tests, test suites and execution
  * Added controllers to backend to receive post from preprocessor
  * Added tests

This issue closes #37 #43 #47

### CODE REVIEW

File: db/changelog/v-1/04-changeset-test-table.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd">
+
+    <changeSet id="add-test-table" author="aktsay" context="dev or prod">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <tableExists tableName="test"/>
+            </not>
+        </preConditions>
+        <createTable tableName="test">
+            <column name="id" type="varchar(255)">
```
Comment (petertrr) [inline db/changelog/v-1/04-changeset-test-table.xml:15]: I believe we should use bigint for all ids now

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ExecutionController.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+/**
+ * Controller that accepts executions
+ */
+@RestController
+class ExecutionController(private val executionService: ExecutionService) {
+    /**
+     * @param execution
+     */
+    @PostMapping("/createExecution")
+    fun createExecution(@RequestBody execution: Execution) {
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ExecutionController.kt:18]: Maybe add `updateExecution` rn too?

File: save-orchestrator/src/main/resources/application-dev.properties [hunk 3]
```diff
@@ -0,0 +1,14 @@
+spring.h2.console.enabled=true
```
Comment (petertrr) [inline save-orchestrator/src/main/resources/application-dev.properties:1]: Is it by mistake? We don't need DB setttings in orchestrator

**Review by petertrr — CHANGES_REQUESTED** [review]

File: save-backend/src/test/kotlin/org/cqfn/save/backend/controller/ExecutionControllerTest.kt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            .uri("/createExecution")
+            .contentType(MediaType.APPLICATION_JSON)
+            .body(BodyInserters.fromValue(execution))
+            .exchange()
+            .expectStatus()
+            .isOk
+
+        val databaseData = executionRepository.findAll()
+
+        assert(databaseData.any { it.status == execution.status && it.id == execution.id })
```
Comment (petertrr) [inline save-backend/src/test/kotlin/org/cqfn/save/backend/controller/ExecutionControllerTest.kt:68]: Use junit or kotlin.test assertions (`assertEquals`, `assertTrue`) instead of JVM-asserts

File: save-backend/src/main/kotlin/org/cqfn/save/backend/service/ExecutionService.kt [hunk 5]
```diff
@@ -15,4 +18,15 @@ class ExecutionService(private val executionRepository: ExecutionRepository) {
     fun saveExecution(execution: Execution) {
         executionRepository.save(execution)
     }
+
+    /**
+     * @param execution
+     */
+    fun updateExecution(execution: ExecutionUpdateDto) {
+        val databaseExecution = executionRepository.findById(execution.id).get()
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/service/ExecutionService.kt:26]: You shouldn't rely on incoming request correctness; therefore you shouldn't call `get()` on optional without aditional checks. If someone tries to update non existing execution, the service should return 4xx error

**Review by petertrr — CHANGES_REQUESTED** [review]

File: db/changelog/v-1/04-changeset-test-table.xml (hunk 1 above)
Comment (aktsay6) [reply db/changelog/v-1/04-changeset-test-table.xml:15]: As I remember this one should be hash from two files.

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ExecutionController.kt (hunk 2 above)
Comment (aktsay6) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ExecutionController.kt:19]: Added

File: save-orchestrator/src/main/resources/application-dev.properties (hunk 3 above)
Comment (aktsay6) [reply save-orchestrator/src/main/resources/application-dev.properties]: Fixed

File: save-backend/src/test/kotlin/org/cqfn/save/backend/controller/ExecutionControllerTest.kt (hunk 4 above)
Comment (aktsay6) [reply save-backend/src/test/kotlin/org/cqfn/save/backend/controller/ExecutionControllerTest.kt]: Fixed

File: save-common/src/commonMain/kotlin/org/cqfn/save/execution/ExecutionUpdateDto.kt [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+/**
+ * @property id
+ * @property status
+ */
+class ExecutionUpdateDto(
+    val id: Long,
+    val status: ExecutionStatus
+) {
+    constructor() : this(0, ExecutionStatus.PENDING)
```
Comment (petertrr) [inline save-common/src/commonMain/kotlin/org/cqfn/save/execution/ExecutionUpdateDto.kt:11]: Why is this needed?

**Review by petertrr — APPROVED** [review]

**Review by petertrr — CHANGES_REQUESTED** [review]
Fix code style

File: save-common/src/jvmMain/kotlin/org/cqfn/save/entities/Execution.kt [hunk 7]
```diff
@@ -11,16 +11,18 @@ import javax.persistence.Enumerated
  * @property endTime
  * @property status
  * @property testSuiteIds
- * @property id
  */
 @Entity
-class Execution(
+data class Execution(
```
Comment (petertrr) [inline save-common/src/jvmMain/kotlin/org/cqfn/save/entities/Execution.kt:16]: I believe we have discussed that data classes shouldn't be used with hibernate?

File: db/changelog/v-1/04-changeset-test-table.xml (hunk 1 above)
Comment (petertrr) [reply db/changelog/v-1/04-changeset-test-table.xml:15]: Yes, right

File: save-common/src/commonMain/kotlin/org/cqfn/save/execution/ExecutionUpdateDto.kt [hunk 8]
```diff
@@ -0,0 +1,12 @@
+package org.cqfn.save.execution
+
+/**
+ * @property id
+ * @property status
+ */
+data class ExecutionUpdateDto(
```
Comment (petertrr) [inline save-common/src/commonMain/kotlin/org/cqfn/save/execution/ExecutionUpdateDto.kt:7]: This should be `@Serializable`

**Review by petertrr — APPROVED** [review]
